python/revng/api/_capi.py

```diff
@@ -39,6 +39,8 @@ def decrement(self):
     def mark_end(self):
         with self.lock:
             self.ending = True
+            if self.counter == 0:
+                self.callback()
```

Comment:
In some situations `rp_shutdown()` wouldn't be called, and that would lead to a crash after pressing `Ctrl+C`.
This was solved by:
* Checking if we already have 0 pointers when calling `mark_end` (as shown here), since on some situations the counter could be already 0
* Reducing the amount of places the `Manager` is passed around
* Forcing the `rp_manager` pointer to be `None` so that the counter drops to 0

Reply:
Added docs